app/Code.gs

```diff
@@ -150,40 +150,50 @@ function asciidocHandleFontStyle(text, offset, distinctContent) {
   var isBold = text.isBold(offset);
   var isItalic = text.isItalic(offset);
   var isUnderline = text.isUnderline(offset);
-  var isPlain = !isBold && !isItalic && !isUnderline;
+  var isStrikethrough = text.isStrikethrough(offset);
+  var htmlBuf = ''
+  // FIXME: getTextAttributeIndices doesn't split on different fonts,
+  // makeing this almost useless
   var isCode = isTextCode(text);
   // Prefix markup
   if (isUnderline) {
-    // Underline doesn't play nice with others
-    result = result + '+++<u>';
-  } else {
-    if (isBold) {
-      result = result + new Array(numOccurence).join('*');
-    }
-    if (isItalic) {
-      result = result + new Array(numOccurence).join('_');
-    }
-    if (isCode) {
-      result = result + '+';
-    }
+    htmlBuf += '<u>'; // or asciidoc.css class: underline
+  }
+  if (isStrikethrough) {
+    htmlBuf += '<s>'; // or asciidoc.css class: line-through
+  }
+  if (htmlBuf !== '') {
+  	result += '+++' + htmlBuf + '+++';
+  	htmlBuf = '';
+  }
+  if (isBold) {
+    result = result + new Array(numOccurence).join('*');
+  }
+  if (isItalic) {
+    result = result + new Array(numOccurence).join('_');
+  }
+  if (isCode) {
+    result = result + '+';
   }
   // Content
-  result = result + distinctContent;
+  result += distinctContent;
   // Suffix markup
-  if (!isUnderline) {
-    if (isItalic) {
-      result = result + new Array(numOccurence).join('_');
-    }
-    if (isBold) {
-      result = result + new Array(numOccurence).join('*');
-    }
-    if (isCode) {
-      result = result + '+';
-    }
-  } else {
-    result = result + '</u>+++';
+  if (isCode) {
+    result = result + '+';
   }
-  return result;
+  if (isItalic) {
+    result = result + new Array(numOccurence).join('_');
+  }
+  if (isBold) {
+    result = result + new Array(numOccurence).join('*');
+  }
+  if (isStrikethrough) {
+    htmlBuf += '</s>';
+  }
+  if (isUnderline) {
+	  htmlBuf += '</u>';
+  }
+  return htmlBuf !== '' ? result + '+++' + htmlBuf + '+++' : result;
```

Comment:
I would prefer to avoid inline:

```javascript
if (htmlBuf !== '') {
  result += '+++' + htmlBuf + '+++';
}
return result;
```

I think this read better since we're doing the exact same steps in reverse order.